src/main/java/org/elasticsearch/index/gateway/none/NoneIndexShardGateway.java

```diff
@@ -53,7 +53,7 @@ public String toString() {
 
     @Override
     public void recover(boolean indexShouldExists, RecoveryState recoveryState) throws IndexShardGatewayRecoveryException {
-        recoveryState.getIndex().startTime(System.currentTimeMillis());
+        indexShard.prepareForStoreRecovery();
```

Comment:
I personally like prepare better and it fits the other method names. I don't feel to strongly about it though so let me know if you do and I'll change it. I did rename to be prepareForIndexRecovery to be inline with all the rest (RecoveryState.getIndex(), Stage.INDEX etc). I like the name store better, but this will be a bigger change have BWC implications.